src/components/EditButton/EditButton.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import { Button } from 'reactstrap';
+import ModalWindow from '../ModalWindow/ModalWindow';
+import './EditButton.scss';
+
+class EditButton extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+    };
+
+    this.openModalWindow = this.openModalWindow.bind(this);
+  }
+
+  toggleModalWindow = () => {
+    this.setState((prev) => ({ ...prev, isOpen: !prev.isOpen }));
+  };
+
+  openModalWindow() {
+    this.setState({
+      isOpen: true,
+    });
+  }
+
+  render() {
+    const { title, body } = this.props;
+    const { isOpen } = this.state;
+
+    return (
+      <>
+        <Button
+          className="EditButton__btn"
+          color="primary"
+          onClick={this.openModalWindow}
+        >
+          Edit
+        </Button>
+        <ModalWindow
```

Comment:
Also please check. If you render 100 buttons. How much dialogs will be created in DOM?

Think about having one dialog component which will render any content inside

Reply:
Similar to that: https://dev.to/zettadam/how-to-handle-multiple-modals-in-a-react-application-2pei